### ISSUE TITLE
Allow access to EntryView properties in query predicates

### ISSUE THREAD

**Issue by mgoldver** [hazelcast/hazelcast#17944]
Need to define a query to get map entries based on the information available in EntryView. For example, get keys or values or entries modified after a specific time. Or, find entries that expire in less than an hour. 
Currently, the only way of doing it (to my knowledge) is to add a timestamps to each object and evaluate it.

The ask: Extend query API to allow EntryView properties to be evaluated as part of the predicate expressions.

**Comment by mmedenjak** [comment]
Hi @mgoldver ! Thank you for the suggestion. I'll add it to the backlog and we'll assess the priority of it. Right now we're heavily invested in the new SQL engine and we're not investing into the query API.

**Comment by aaronriekenberg** [comment]
Is there a way to access EntryView properties (ie creationTime, expirationTime, accessTime) from SQL?  Thanks.

**Comment by mmedenjak** [comment]
@aaronriekenberg I don't think so. At least not yet. But we can consider it in the future cc @viliam-durina
